### PR TITLE
Jetpack connect: Remove authorize wrapper

### PR DESCRIPTION
**Pull request by sirreal**
This PR removes the `authorize-form` wrapper which contained logged-in (authorize) and logged-out (signup) components, displaying one or the other based on whether the user is logged in or not.

The parent component only adds complexity and obfuscation. This PR removes it.

The intention was to maintain nearly identical functionality, structure, behavior, etc. The only place a slight deviation from absolutely identical behavior should be observed is the removal of one `<div>` in one of the new components. This should have no effects on the exprience.

* No visible differences
* No functional differences (except for that `div`)

## Testing
* Hit the auth step of the flowz 😎

### CODE REVIEW

File: client/jetpack-connect/controller.js [hunk 1]
```diff
@@ -205,6 +206,39 @@ export function authorizeForm( context, next ) {
 	next();
 }
 
+export function authorizeForm( context, next ) {
+	analytics.pageView.record( 'jetpack/connect/authorize', 'Jetpack Authorize' );
+
+	removeSidebar( context );
+
+	const { query } = context;
+	const validQueryObject = validator( authorizeQueryDataSchema )( query );
```
Comment (sirreal) [inline client/jetpack-connect/controller.js]: This is now duplicated. We could extract to a helper function but I prefer to leave that for a follow-up PR which could focus on that. This PR is already quite extensive.

File: client/jetpack-connect/authorize-form.jsx [hunk 2]
```diff
@@ -1,67 +0,0 @@
...
-		isLoggedIn: PropTypes.bool.isRequired,
-		recordTracksEvent: PropTypes.func.isRequired,
-	};
-
-	componentWillMount() {
-		this.props.recordTracksEvent( 'calypso_jpc_authorize_form_view' );
-	}
-
-	handleClickHelp = () => {
-		this.props.recordTracksEvent( 'calypso_jpc_help_link_click' );
```
Comment (seear) [inline client/jetpack-connect/authorize-form.jsx]: Do we lose this event now? Maybe check with support to see if they are using it.

File: client/jetpack-connect/controller.js [hunk 3]
```diff
@@ -91,10 +92,10 @@ const getPlanSlugFromFlowType = ( type, interval = 'yearly' ) => {
 	return get( planSlugs, [ interval, type ], '' );
 };
 
-export function redirectWithoutLocaleifLoggedIn( context, next ) {
+export function redirectWithoutLocaleIfLoggedIn( context, next ) {
```
Comment (seear) [inline client/jetpack-connect/controller.js]: Was staring at this for ages trying to spot the difference :D

File: client/jetpack-connect/authorize-form.jsx (hunk 2 above)
Comment (sirreal) [reply client/jetpack-connect/authorize-form.jsx]: No, this was actually [should have moved](https://github.com/Automattic/wp-calypso/pull/20406/files#diff-13f3d900556a872d3fa814d9ef4d18aeR26) in #20406. Instead it was copied. Here it no longer serves, but the event is maintained.

**Pull request merged**